### PR TITLE
Sa/list integration

### PR DESCRIPTION
**Pull request by SultanArshad**

### CODE REVIEW

File: app-kotlin/build.gradle [hunk 1]
```diff
@@ -1,47 +1,95 @@
...
-  kotlinOptions {
-    jvmTarget = '1.8'
+  sourceSets {
+    main.java.srcDirs += 'src/main/kotlin'
   }
 }
 
 dependencies {
-  implementation 'androidx.core:core-ktx:1.7.0'
+  implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.7.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:41]: do we need this?

File: app-kotlin/build.gradle [hunk 2]
```diff
@@ -1,47 +1,95 @@
...
+
+//  Conductor
+  implementation 'com.bluelinelabs:conductor:3.1.5'
+  implementation 'com.bluelinelabs:conductor-archlifecycle:3.1.5'
+
+// AndroidX Transition change handlers:
+  implementation 'com.bluelinelabs:conductor-androidx-transition:3.1.5'
+
+// ViewPager PagerAdapter:
+  implementation 'com.bluelinelabs:conductor-viewpager:3.1.5'
```
Comment (xxfast) [inline app-kotlin/build.gradle:56]: do we need this?

File: app-kotlin/build.gradle [hunk 3]
```diff
@@ -1,47 +1,95 @@
...
+
+// ViewPager2 Adapter:
+  implementation 'com.bluelinelabs:conductor-viewpager2:3.1.5'
```
Comment (xxfast) [inline app-kotlin/build.gradle:59]: do we need this?

File: app-kotlin/build.gradle [hunk 4]
```diff
@@ -1,47 +1,95 @@
...
+
+// RxJava2 Autodispose support:
+  implementation 'com.bluelinelabs:conductor-autodispose:3.1.5'
```
Comment (xxfast) [inline app-kotlin/build.gradle:62]: do we need this?

File: app-kotlin/build.gradle [hunk 5]
```diff
@@ -1,47 +1,95 @@
...
-  implementation 'com.google.android.material:material:1.6.1'
-  implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+
+//  Support libraries
+  implementation 'androidx.appcompat:appcompat:1.4.2'
+  implementation 'androidx.recyclerview:recyclerview:1.2.1'
+
+//  Conductor
+  implementation 'com.bluelinelabs:conductor:3.1.5'
+  implementation 'com.bluelinelabs:conductor-archlifecycle:3.1.5'
```
Comment (xxfast) [inline app-kotlin/build.gradle:50]: duplicated dependency

File: app-kotlin/build.gradle [hunk 6]
```diff
@@ -1,47 +1,95 @@
...
+  implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.3'
+  implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.3'
+
+//  picasso
+  implementation 'com.squareup.picasso:picasso:2.8'
+
+//  LifeCycle
+  implementation 'androidx.lifecycle:lifecycle-common:2.5.0'
+  implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.0'
+  implementation 'androidx.lifecycle:lifecycle-livedata-ktx:2.5.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:86]: do we need this?

File: app-kotlin/build.gradle [hunk 7]
```diff
@@ -1,47 +1,95 @@
...
+
+  //viewModel
+  implementation 'androidx.lifecycle:lifecycle-extensions:2.2.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:89]: do we need this?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package au.com.gridstone.trainingkotlin
+
+object Constants {
+  const val MY_TAG = "MY_LOG_TAG"
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt:4]: is this used?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+  //  const val  BASE_URL = "https://pokeapi.co/api/v2/pokemon/"
+  const val BASE_URL = "https://httpstat.us/400/"
+  const val IMAGE_URL =
+    "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork"
+}
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt:10]: These should belong in build configs

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/LoggingInterceptor.kt [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+object LoggingInterceptor : Interceptor {
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val request: Request = chain.request()
+    println(request)
+    val response: Response = chain.proceed(request)
+    println(response)
+    return response
+  }
+
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/LoggingInterceptor.kt:15]: nit: extra white space

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.changehandler.VerticalChangeHandler
+import com.squareup.picasso.Picasso
+
+internal class HomeListAdapter(private var itemsList: List<Pokemon>, private val router: Router) :
+  RecyclerView.Adapter<HomeListAdapter.MyViewHolder>() {
+  internal inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    var itemView: FrameLayout = view.findViewById(R.id.view)
+    var image: ImageView = view.findViewById(R.id.image_list)
+    var name: TextView = view.findViewById(R.id.name)
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:26]: why `var`?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import au.com.gridstone.trainingkotlin.views.DetailViewController
+import com.bluelinelabs.conductor.Router
+import com.bluelinelabs.conductor.RouterTransaction
+import com.bluelinelabs.conductor.changehandler.VerticalChangeHandler
+import com.squareup.picasso.Picasso
+
+internal class HomeListAdapter(private var itemsList: List<Pokemon>, private val router: Router) :
+  RecyclerView.Adapter<HomeListAdapter.MyViewHolder>() {
+  internal inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:23]: why `inner`?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+  @NonNull
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+    val itemView = LayoutInflater.from(parent.context)
+      .inflate(R.layout.item_home_recyclerview, parent, false)
+    return MyViewHolder(itemView)
+  }
+
+  override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+    val item = itemsList[position]
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:37]: explicit type here

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+      .load("$IMAGE_URL/$imageNumbers.png")
+      .placeholder(R.drawable.ic_placeholder)
+      .into(holder.image)
+  }
+
+  override fun getItemCount(): Int {
+    return itemsList.size
+  }
+
+  fun setList(it: PokemonResults?) {
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:58]: why is this taking in a nullable value? 🤔

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+  ): Response<PokemonResults>
+
+  @GET("{pokemonUrl}")
+  suspend fun getPokemonDetails(@Path("url") url: String): Response<PokemonData>
+}
+
+
+//  @POST("parts") fun addPartAsync(@Body newPart : PartData): Deferred<Response<Void>>
+//  @DELETE("parts/{id}") fun deletePartAsync(@Path("id") id: Long) : Deferred<Response<Void>>
+//  @PUT("parts/{id}") fun updatePartAsync(@Path("id") id: Long, @Body newPart: PartData) : Deferred<Response<Void>>
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt:21]: no need to commit in commented out code 😄

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt [hunk 16]
```diff
@@ -0,0 +1,19 @@
+package au.com.gridstone.trainingkotlin.data
+
+import com.squareup.moshi.JsonClass
+
+// mapping jason top object
+@JsonClass(generateAdapter = true)
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt:6]: you can avoid all the JsonAdapter generation, if we use the `KotlinJsonAdapter()`
```kotlin
Moshi.Builder()
  .add(KotlinJsonAdapterFactory())
  .build() 
```

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/HomeViewController.kt [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+  override fun onAttach(view: View) {
+    val recyclerView: RecyclerView = view.findViewById(R.id.home_recyclerView)
+    val progressBar: ProgressBar = view.findViewById(R.id.home_progressBar)
+    val errorImage: ImageView = view.findViewById(R.id.image_error)
+    val layoutManager = LinearLayoutManager(applicationContext)
+    recyclerView.layoutManager = layoutManager
+    val homeListAdapter = HomeListAdapter(emptyList(), router)
+    recyclerView.adapter = homeListAdapter
+    GlobalScope.launch(Dispatchers.Main) {
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/HomeViewController.kt:42]: Avoid `GlobalScope`. Instead, scope this launched coroutine to the controller scope

**Review by xxfast — CHANGES_REQUESTED** [review]
Nice work 👍

File: app-kotlin/build.gradle (hunk 1 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:41]: yeah we don't need this cleaning  thanks

File: app-kotlin/build.gradle (hunk 2 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:56]: removed

File: app-kotlin/build.gradle (hunk 5 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:50]: thanks for mentioning resolved.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt (hunk 14 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:58]: it was before when I had sent data without implementing the sealed classes artitechure. no by going through the code and by dry running, I realize that it won't need this nullable anymore thanks for mentioning I had removed this.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt (hunk 15 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt:21]: sure keep in mind next time

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt (hunk 16 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt:6]: Sure

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/HomeViewController.kt (hunk 17 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/views/HomeViewController.kt:42]: thanks @xxfast  well presented and explained cheers.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt (hunk 9 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt:10]: thanks i will be doing this next after implementing detail screen by doing some research over internet

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt (hunk 8 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt:4]: yes this is used by the NetworkConnectionInterceptor class to log network states

File: app-kotlin/build.gradle [hunk 18]
```diff
@@ -1,47 +1,72 @@
...
+  implementation 'com.bluelinelabs:conductor-archlifecycle:3.1.5'
+
+//  For retrofit / networking
+  implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+  implementation 'com.squareup.retrofit2:converter-moshi:2.9.0'
+  implementation 'com.squareup.okhttp3:okhttp:5.0.0-alpha.1'
+
+//  Moshi
+  implementation 'com.squareup.moshi:moshi:1.13.0'
+  kapt 'com.squareup.moshi:moshi-kotlin-codegen:1.13.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:60]: no longer needed i think

File: app-kotlin/build.gradle [hunk 19]
```diff
@@ -1,47 +1,72 @@
...
   dependencies {
-    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.0'
+    classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.10'
   }
 }
 
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-kapt'
+apply plugin: 'org.jetbrains.kotlin.android'
```
Comment (xxfast) [inline app-kotlin/build.gradle:15]: no longer needed i think

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    val imageNumbers: String = item.url.split("/").dropLast(1).last()
+    Picasso.get()
+      .load("$IMAGE_URL/$imageNumbers.png")
+      .placeholder(R.drawable.ic_placeholder)
+      .into(holder.image)
+  }
+
+  override fun getItemCount(): Int {
+    return itemsList.size
+  }
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:52]: use expression syntax

**Review by xxfast — CHANGES_REQUESTED** [review]
Its getting there 👍

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt (hunk 9 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/Constants.kt:10]: Thanks i had fixed this.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.RecyclerView
+import au.com.gridstone.trainingkotlin.Constants.IMAGE_URL
+import au.com.gridstone.trainingkotlin.R
+import au.com.gridstone.trainingkotlin.data.PokemonResults
+import au.com.gridstone.trainingkotlin.data.Pokemon
+import com.bluelinelabs.conductor.Router
+import com.squareup.picasso.Picasso
+
+internal class HomeListAdapter(private var itemsList: List<Pokemon>, private val router: Router) :
```
[the remark this answers is not in the document]
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:17]: We need this as var as we a setting the value after getting data from network as we discussed thanks for the explanation

File: app-kotlin/build.gradle (hunk 18 above)
Comment (SultanArshad) [reply app-kotlin/build.gradle:60]: Thanks for mentioning yes we don't need this anymore

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt (hunk 20 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:52]: thanks fixed

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+
+  single<PokemonService> {
+    get<Retrofit>().create(PokemonService::class.java)
+  }
+
+  single { HomeViewModel() }
+
+  scope(named(SCOPE_DETAIL)) {
+    scoped { (id: String) -> DetailViewModel(id) }
+  }
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt:51]: is this intended? 🤔

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+  override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+    val item: Pokemon = itemsList[position]
+    holder.name.text = item.name
+    loadImage(holder, item)
+    holder.itemView.setOnClickListener {
+      router.pushController(RouterTransaction.with(
+        DetailViewController(item))
+        .pushChangeHandler(FadeChangeHandler())
+        .popChangeHandler(FadeChangeHandler())
+      )
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:45]: is this intended? 🤔

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt [hunk 24]
```diff
@@ -0,0 +1,14 @@
...
+import au.com.gridstone.trainingkotlin.data.PokemonData
+import retrofit2.Response
+import retrofit2.http.*
+
+interface  PokemonService {
+  @GET("?limit=151")
+  suspend fun getPokemons(): Response<PokemonResults>
+
+  @GET("{pokemonId}")
+  suspend fun getPokemonDetails(@Path("pokemonId") pokemonId: String): Response<PokemonData>
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt:13]: is this intended? 🤔

**Review by xxfast — CHANGES_REQUESTED** [review]
Looks like some code from your other PR creeped in to this PR.

File: app-kotlin/build.gradle [hunk 25]
```diff
@@ -1,47 +1,74 @@
...
   }
 }
 
 dependencies {
-  implementation 'androidx.core:core-ktx:1.7.0'
   implementation 'androidx.appcompat:appcompat:1.4.2'
-  implementation 'com.google.android.material:material:1.6.1'
-  implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+
+//  Support libraries
```
[the remark this answers is not in the document]
Comment (SultanArshad) [reply app-kotlin/build.gradle:48]: Sure thanks for the information

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt (hunk 22 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt]: sorry was not my intention to commit this I will be aware next time thanks for pointing this out.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+  single<Retrofit> {
+    Retrofit.Builder()
+      // .baseUrl(Constants.BASE_URL)
+      .baseUrl(BASAE_URL)
+      .client(get())
+      // Moshi maps JSON to classes
+      .addConverterFactory(MoshiConverterFactory.create(
+        Moshi.Builder()
+          .add(KotlinJsonAdapterFactory())
+          .build()))
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt:27]: declare moshi as its own leaf in our dependency graph
```kotlin
single { 
  Moshi.Builder()
    .add(KotlinJsonAdapterFactory())
    .build()))
}

single<Retrofit> {
  Retrofit.Builder()
      .baseUrl(BASAE_URL)
      .client(get())
      .addConverterFactory(MoshiConverterFactory.create(get())
}      
```

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+import org.koin.core.context.GlobalContext.startKoin
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+val applicationModule: Module = module {
+  single<Retrofit> {
+    Retrofit.Builder()
+      // .baseUrl(Constants.BASE_URL)
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt:20]: commented out code?

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+
+  private fun loadImage(holder: MyViewHolder, item: Pokemon) =
+    Picasso.get()
+      .load("$IMAGE_URL/${item.id}.png")
+      .placeholder(R.drawable.ic_placeholder)
+      .into(holder.image)
+
+  override fun getItemCount(): Int = itemsList.size
+
+  fun setList(it: PokemonResults) {
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:52]: if the signature reads `setList` it should probably take in a list 🤔

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+data class PokemonResults(val results: List<Pokemon>)
+
+@Parcelize
+data class Pokemon(
+  val url: String,
+  val name: String,
+  val id: String = url.split("/").dropLast(1).last(),
+) : Parcelable
+
+data class PokemonData(val stats: List<Stats>)
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt:15]: is this used?

File: app-kotlin/build.gradle [hunk 30]
```diff
@@ -1,47 +1,74 @@
...
+
+//  Kotlin Coroutines
+  implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.3'
+
+//  picasso
+  implementation 'com.squareup.picasso:picasso:2.8'
+
+  // Koin for Kotlin apps
+  implementation 'io.insert-koin:koin-core:3.2.0'
+  implementation 'io.insert-koin:koin-android:3.2.0'
```
Comment (xxfast) [inline app-kotlin/build.gradle:73]: nit: formatting is bit weird here

**Review by xxfast — CHANGES_REQUESTED** [review]
Almost there 👍

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt (hunk 26 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/MainApplication.kt:27]: Thanks i ahd updated this with separate definition

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt (hunk 28 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/adapters/HomeListAdapter.kt:52]: sure  i set the signature to setPokemonResults

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt (hunk 29 above)
Comment (SultanArshad) [reply app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/data/DataModel.kt:15]: no in this pr  it is used by the next pr, sorry about that i am getting used to this now will keep this inn mind next time.

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt [hunk 31]
```diff
@@ -0,0 +1,10 @@
+package au.com.gridstone.trainingkotlin.apiservices
+
+import au.com.gridstone.trainingkotlin.data.PokemonResults
+import retrofit2.Response
+import retrofit2.http.*
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/apiservices/PokemonService.kt:5]: According to the style guide, [avoid wildcard imports](https://github.com/Gridstone/KotlinStyleGuide#imports)

File: app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/viewmodels/HomeViewModel.kt [hunk 32]
```diff
@@ -0,0 +1,43 @@
+package au.com.gridstone.trainingkotlin.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.gridstone.trainingkotlin.apiservices.PokemonService
+import au.com.gridstone.trainingkotlin.data.PokemonResults
+import au.com.gridstone.trainingkotlin.viewmodels.HomeViewState.Loading
+import kotlinx.coroutines.*
```
Comment (xxfast) [inline app-kotlin/src/main/kotlin/au/com/gridstone/trainingkotlin/viewmodels/HomeViewModel.kt:8]: [avoid wildcard imports](https://github.com/Gridstone/KotlinStyleGuide#imports)

**Review by xxfast — APPROVED** [review]
Nice work 👍

**Pull request merged**